src/main/java/mate/academy/spring/security/CinemaUserDetailsService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CinemaUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CinemaUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(
+                () -> new UsernameNotFoundException("User with username (" + username
+                        + ") is not registered"));
+        UserBuilder builder = withUsername(username);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream().map(Role::getRoleName).toArray(String[]::new));
```

Comment:
does the code work corectrly with accessing to different endpoints for different roles?
if not, consider this change
```suggestion
        builder.roles(user.getRoles().stream().map(Role::getRoleName).toArray(String[]::new));
```

Reply:
Yeah, thx. I misunderstood the description of the method.
![image](https://github.com/mate-academy/jv-spring-security-part-2/assets/96497090/967e63aa-0593-4e34-82e0-28e55883661a)